Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -359,32 +365,34 @@ class TabTrayController: UIViewController, UITabBarDelegate, UICollectionViewDel
     var profile: Profile!
     var numberOfColumns: Int!
 
-    var navBar: UINavigationBar!
+    var navBar: UIView!
     var addTabButton: UIButton!
     var settingsButton: UIButton!
 
-    func SELstatusBarFrameWillChange(notification: NSNotification) {
-        self.view.setNeedsLayout()
+    var statusBarFrame: CGRect {
+        return UIApplication.sharedApplication().statusBarFrame
+    }
+
+    var collectionViewTransitionSnapshot: UIView?
+
+    func SELstatusBarFrameDidChange(notification: NSNotification) {
+        self.view.setNeedsUpdateConstraints()
     }
 
     deinit {
-        NSNotificationCenter.defaultCenter().removeObserver(self, name: UIApplicationWillChangeStatusBarFrameNotification, object: nil)
```

Comment:
I think this is fine, but maybe it is better to use `viewWillAppear` and `viewWillDisappear` to register/unregister these?